pymatgen/analysis/graphs.py

```diff
@@ -369,6 +382,7 @@ def add_edge(
             return
 
         # sanitize types
+        # TODO(@DanielYang59) fix mypy error
```

Comment:
Not sure why I'm getting this following error:
```
pymatgen/analysis/graphs.py:386: error: Incompatible types in assignment (expression has type "tuple[int, ...]", variable has type "tuple[int, int, int]")  [assignment]
pymatgen/analysis/graphs.py:386: error: Incompatible types in assignment (expression has type "tuple[int, ...]", variable has type "Optional[tuple[int, int, int]]")  [assignment]
```

A simple code to recreate this:
```python
t: tuple[int, int, int] = (1, 0, 0)

t = tuple(map(int, t))
```
would trigger the same error.